.github/workflows/test.yml

```
@@ -14,14 +14,18 @@ jobs:
       matrix:
         python: ["3.11", "3.10", "3.9", "3.8"]
         django: ["django>=3.2,<3.3", "django>=4.1,<4.2", "django>=4.2,<4.3"]
-        wagtail: ["wagtail>=4.1,<4.2", "wagtail>=4.2,<5.0"]
+        wagtail: ["wagtail>=4.1,<4.2", "wagtail>=4.2,<5.0", "wagtail>=5.0,<5.1"]
```

If we want to support Wagtail 5.1, shouldn't we include version less than 5.2?
```suggestion
        wagtail: ["wagtail>=4.1,<4.2", "wagtail>=4.2,<5.0", "wagtail>=5.0,<5.2"]
```

---

It looks like the maintainer specifies the latest in the include below so I updated it there. I think if it's in this matrix it will run twice. Actually I did add it there at first until I worked out what was happening below. Do you agree?

---

I've just had a look at the [Github Actions docs](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) and it seems as though if Wagtail 5.1 were included in both the test matrix and the include, it would only be tested once.
As it stands, with Wagtail 5.1 only in the included section, it means that it will only be tested against Python 3.11 and Django 4.2, but with the important addition of setting "latest:true".
If we're content with only testing Wagtail 5.1 against the latest versions of python and django, then as it stands with 5.1 only in the include section would be fine. 
If however, we'd also like to test its compatibility across the whole matrix of Python and Django versions, we should also set it in the matrix.
Presuming that the setting of "latest:true" is desirable, we could include it in both the matrix and keep it in the include section. If we did this, it would run all combinations of Python:Django:Wagtail versions, and on the run of Python3.11:Django4.2:Wagtail5.1, latest would be set to true (this would only run once).
I think this last case might be the best option, as it also keeps it really clear to maintainers exactly what is being tested, still sets "latest:true" and also tests Wagtail 5.1 with each version of Python and Django.
Let me know what you think

---

Thanks @lparsons396 this is some deep investigation. 👍 

My understanding is the overall jobs that will be run is calculated by the matrix for all permutations, then minus the excludes, then plus the include.
Only one version of py,dj,wt will run on each job.
It looks like the suggestion means 5.0 isn't tested as it install 5.1 here `"wagtail>=5.0,<5.2"`

Although it's hard to see the effect without clicking on each job. This job: https://github.com/torchbox-forks/wagtail-jotform/actions/runs/6419089918/job/17428203790#step:4:16 installs 5.1.2 and misses 5.0.

I made a couple of draft pr's to try and view the difference: https://github.com/torchbox-forks/wagtail-jotform/pull/3 and https://github.com/torchbox-forks/wagtail-jotform/pull/2

---

Yeah, this is really interesting! I'm convinced :) thanks for your clarification